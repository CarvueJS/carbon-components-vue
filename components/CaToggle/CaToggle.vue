<template>
  <div class="bx--form-item">
    <input class="bx--toggle" id="toggle1" type="checkbox">
    <label class="bx--toggle__label" for="toggle1">
      <span class="bx--toggle__text--left">{{this.left}}</span>
      <span class="bx--toggle__appearance"></span>
      <span class="bx--toggle__text--right">{{this.right}}</span>
    </label>
  </div>
</template>

<script>
  export default {
    name: 'ca-toggle',
    props: {
      left: {
        type: String,
        default: 'Off',
      },
      right: {
        type: String,
        default: 'On',
      },
    },
  };
</script>

<style></style>
