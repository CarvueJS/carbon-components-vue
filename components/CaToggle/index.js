import CaToggle from './CaToggle.vue';

CaToggle.install = (Vue) => {
  Vue.component(CaToggle.name, CaToggle);
};

export default CaToggle;
