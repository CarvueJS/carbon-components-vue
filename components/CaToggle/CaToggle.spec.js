import { shallow } from 'vue-test-utils';
import CaToggle from './CaToggle.vue';

describe('CaToggle', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaToggle);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
