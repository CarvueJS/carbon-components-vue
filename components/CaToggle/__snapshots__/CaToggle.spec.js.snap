// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaToggle Mathes snapshot 1`] = `
<div class="bx--form-item">
  <input id="toggle1" type="checkbox" class="bx--toggle">
  <label for="toggle1" class="bx--toggle__label"><span class="bx--toggle__text--left">Off</span> <span class="bx--toggle__appearance"></span> <span class="bx--toggle__text--right">On</span></label>
</div>
`;
